swift/ios-shoppe-demo/Model/Order.swift

```diff
@@ -11,13 +11,20 @@ import Foundation
 class Order {
     var items: [Product] = []
     var date: Date = Date()
+    var orderId: String = makeId(len: 9)
+    var shipping: Double = 5.99
+    var tax: Double = 2.85
     var cartOrderTotal: Double {
         return items.reduce(0, { $0 + getTotal(for: $1) })
     }
 
     func addProduct(_ productName: String ) {
         items.filter { $0.title == productName }.first?.quantity += 1
     }
+    
+    func getProduct(_ productName: String ) -> Product? {
+        return items.filter { $0.title == productName }.first
+    }
```

Comment:
I would recommend changing this class to use a `[String: Product]` for `items`.

This would enable you to simplify these methods to:

```suggestion
    func addProduct(_ productName: String ) {
        items[productName]?.quantity += 1
    }
    
    func getProduct(_ productName: String ) -> Product? {
        return items[productName]
    }
```

If you don't go with this approach, it would be a good idea to have `addProduct` call `getProduct` rather than it doing its own searching.

Reply:
I _think_ dictionaries is a bit more expensive to loop through than array. `order` item in this case holds all the products, regardless of it's in the cart or not. So I'm inclined to use array. But I did refactored some stuff into functions in `Order`, see file changed.